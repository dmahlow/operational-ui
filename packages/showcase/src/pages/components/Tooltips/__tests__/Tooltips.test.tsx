import * as React from "react"
import { render } from "enzyme"

import TooltipsPage from "../Tooltips"

describe("Tooltips Page", () => {
  it("Should render correctly", () => {
    expect(render(<TooltipsPage />)).toMatchSnapshot()
  })
})
