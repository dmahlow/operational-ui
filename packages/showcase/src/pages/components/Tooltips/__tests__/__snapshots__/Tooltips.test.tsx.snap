// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltips Page Should render correctly 1`] = `
<div>
  <div
    class="css-by30gj"
  >
    Tooltips
  </div>
  <p>
    These elements give helpful hints about actions an end-user can perform. They are designed to be reusable, elegant and unobtrusive. Tooltips are great for UX. They can (and in many cases should) be used along with, and in, other components. For this reason, we took a highly versatile approach to our tooltips.
  </p>
  <h4>
    Usage
  </h4>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-1kfhvsj"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          color
        </td>
        <td>
          A hex value or a named color from your theme.
        </td>
        <td>
          The greys["100"] in your theme.
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          active
        </td>
        <td>
          Is the tooltip currently visible? This is toggle onMouseEnter of the Tooltip's parent
        </td>
        <td>
          false
        </td>
        <td>
          boolean
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          anchor
        </td>
        <td>
          Should the tooltip be anchored to the middle or bottom of its parent? For bottom anchoring, mouse over the guy on the lower left of this page.
        </td>
        <td>
          middle
        </td>
        <td>
          middle | bottom
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          <em>
            children
          </em>
        </td>
        <td>
          What would you like to say in the tooltip?
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="css-by30gj"
  >
    withTooltip
  </div>
  <p>
    Tooltips come with an additional higher-order component (or HOC) called \`withTooltip\`. This component wraps any pre-existing React component that you may have and gives it a tooltip, which you can pass as a prop.
  </p>
  <div
    style="margin-bottom:32px;"
  />
  <h4>
    Usage
  </h4>
  <pre
    style="display:block;overflow-x:auto;padding:0.5em;background:#F0F0F0;color:#444;"
  >
    <code>
      

      <span
        style="font-weight:bold;"
      >
        const
      </span>
       MyDiv = 
      <span>
        <span>
          ()
        </span>
         =&gt;
      </span>
       (
      <span>
        <span>
          &lt;
          <span
            style="font-weight:bold;"
          >
            div
          </span>
          &gt;
        </span>
        Hover for Spanish!
        <span>
          &lt;/
          <span
            style="font-weight:bold;"
          >
            div
          </span>
          &gt;
        </span>
      </span>
      )

      <span
        style="font-weight:bold;"
      >
        const
      </span>
       MyDivWithTooltip = withTooltip(MyDiv)

ReactDOM.render(
  
      <span>
        <span>
          &lt;
          <span
            style="font-weight:bold;"
          >
            MyDivWithTooltip
          </span>
          
    
          <span>
            tooltip
          </span>
          =
          <span
            style="color:#880000;"
          >
            "Hola, compadre!"
          </span>
          
    
          <span>
            tooltipColor
          </span>
          =
          <span
            style="color:#880000;"
          >
            "#f00"
          </span>
          
    
          <span>
            tooltipAnchor
          </span>
          =
          <span
            style="color:#880000;"
          >
            "bottom"
          </span>
          
  &gt;
        </span>
        
    Hover for Spanish
  
        <span>
          &lt;/
          <span
            style="font-weight:bold;"
          >
            MyDivWithTooltip
          </span>
          &gt;
        </span>
      </span>
      ,
  
      <span
        style="color:#397300;"
      >
        document
      </span>
      .querySelector(
      <span
        style="color:#880000;"
      >
        '#app)
)

      </span>
    </code>
  </pre>
  <p>
    <strong>
      Note: The \`color\` and \`anchor\` props are still available, but prepended with the word \`tooltip\` in order not to override any similar props on your component.
    </strong>
  </p>
</div>
`;
